src/main/java/us/kbase/auth2/providers/OrcIDIdentityProviderFactory.java

```diff
@@ -228,6 +233,58 @@ private OrcIDAccessTokenResponse(
 				this.accessToken = accessToken.trim();
 				this.fullName = fullName == null ? null : fullName.trim();
 				this.orcID = orcID.trim();
+				this.idToken = idToken == null ? null : idToken.trim();
+				this.mfaAuthenticated = parseAmrClaim(this.idToken);
+			}
+			
+			/**
+			 * Parses the Authentication Method Reference (AMR) claim from an OpenID Connect ID token
+			 * to determine if multi-factor authentication was used.
+			 * 
+			 * @param idToken the JWT ID token from ORCID
+			 * @return true if MFA was used, false if password only, null if unknown or parsing failed
+			 */
+			private Boolean parseAmrClaim(final String idToken) {
+				if (idToken == null || idToken.trim().isEmpty()) {
+					return null;
+				}
+				
+				try {
+					// JWT format: header.payload.signature
+					final String[] parts = idToken.split("\\.");
+					if (parts.length != 3) {
+						// Invalid JWT format
+						return null;
+					}
```

Comment:
We should assume that OrcID isn't handing out malformed tokens and throw a RuntimeError if this happens.

Reply:
:+1: